### PR TITLE
storagenode: Add more test assertions

### PR DESCRIPTION
**Pull request by ifraixedes**
What: Add more test cases for empty piece ID.

Why:
Because we have a reported issue from one Storage Node Operator which cannot list the unsent order limits and it looks that the issue is that there is one order limit in the DB which has an empty piece ID.

The new tests pass without any change in the implementation but it's good to have them.

The goal was trying to find out the root of the problem of the ticket [V3-2400](https://storjlabs.atlassian.net/browse/V3-2400) but I ended up also adding a bit more assertions not related to it.

Please describe the tests: Added more assertions to current tests, there is no change in the implementation so there are no new tests.
 
Please describe the performance impact: none.

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: storagenode/piecestore/endpoint_test.go [hunk 1]
```diff
@@ -233,6 +238,15 @@ func TestDownload(t *testing.T) {
 			pieceID: orderLimit.PieceId,
 			action:  pb.PieceAction_GET,
 		},
+		{ // should err with piece ID not specified
```
Comment (mobyvb) [inline storagenode/piecestore/endpoint_test.go]: If you do not define `pieceID`, won't it default to `storj.PieceID{}` as the 0 value? Is there a reason to have separate cases that set `pieceID: storj.PieceID{}` and that don't set `pieceID` at all?

Comment (ifraixedes) [reply storagenode/piecestore/endpoint_test.go]: No there isn't is about me being stupidly thinking that they would be `nil`.

I'll remove them

Comment (mobyvb) [reply storagenode/piecestore/endpoint_test.go]: I think it is fine if you want to keep them to be safe. This kind of behavior confuses me still so I am happy with the extra case

Comment (ifraixedes) [reply storagenode/piecestore/endpoint_test.go]: I'm not fine with them, they have duplicated test cases.
It was a mistake.
